packages/template-retail-react-app/app/pages/product-detail/index.test.js

```diff
@@ -289,15 +293,26 @@ describe('Recommended Products', () => {
 })
 
 describe('product bundles', () => {
+    let hasUpdatedBundleChildren = false
     beforeEach(() => {
+        hasUpdatedBundleChildren = false
         global.server.use(
             // Use product bundle instead of product set
             rest.get('*/products/:productId', (req, res, ctx) => {
                 return res(ctx.delay(0), ctx.status(200), ctx.json(mockProductBundle))
             }),
             // For adding items to basket
             rest.post('*/baskets/:basketId/items', (req, res, ctx) => {
-                return res(ctx.json(basketWithProductSet))
+                const basketWithBundle = {
+                    ...basketWithProductBundle,
+                    productItems: bundleProductItemsForPDP
+                }
+                return res(ctx.json(basketWithBundle))
+            }),
+            // Follow up call to update child bundle variant selections
+            rest.patch('*/baskets/:basketId/items', (req, res, ctx) => {
+                hasUpdatedBundleChildren = true
+                return res(ctx.json({}))
```

Comment:
updatedItemsInBasket returns basket with updated items. Make sure we also mock the correct returned data since it will affect the cache machenism in react-query. https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-baskets-v2?meta=updateItemsInBasket